public/script.js

```diff
@@ -1,216 +1,311 @@
+  $('#map').html('');// Empty map when a new animation occurs
+  const timeNow = new moment();
+  $('#startingText').html(`Starting Date (1979-${timeNow.year()}):`);
+  $('#endingText').html(`Ending Date (1979-${timeNow.year()}):`);
 
-async function init(){
-    $('input:radio[name=ext-con]').val(['extent']);//Default values
-    $('input:radio[name=n-s]').val(['n']);
-    $('input:radio[name=dates]').val(['Daily']);
-    document.querySelector('input[name="sDate"]').value = "1990-01-01";
-    document.querySelector('input[name="eDate"]').value = "2018-01-01";
-    $("#map").html("");//Empty map when a new animation occurs
+  projection = getProjection();
+  map = getMap(projection);
+  map.addLayer(createLayer());
+  await updateState();
 
-    
-    projection = getProjection();
-    map = getMap(projection);
-    map.addLayer(createLayer());
-    await updateState();
-
-    STATE.current = moment(STATE.start);
-    map.getLayers().getArray()[0].setZIndex(1000);//loading on top
-    var zoomToExtentControl = new ol.control.ZoomToExtent({
-        extent: getLocationParams().extent,
-        size: [10, 10]
-    });
-    map.addControl(zoomToExtentControl);//Add control to reset view
+  STATE.current = moment(STATE.start);
+  map.getLayers().getArray()[0].setZIndex(1000);// loading on top
+  const zoomToExtentControl = new ol.control.ZoomToExtent({
+    extent: getLocationParams().extent,
+    size: [10, 10],
+  });
+  map.addControl(zoomToExtentControl);// Add control to reset view
 
-    
 
-    STATE.stop = true;
-    $("#pauseAnimation").html("Start Animation");
-    $("#date").html(STATE.current.format("YYYY-MM-DD"));
-    animationLoop();
+  STATE.stop = true;
+  $('#pauseAnimation').html('Start Animation');
+  $('#date').html(STATE.current.format('YYYY-MM-DD'));
+  animationLoop();
 }
 
-
-async function animationLoop(){
-    while (true) {
-        if (!STATE.stop) {
-            nextDate();
-            var wmsParams = {
-                LAYERS: "NSIDC:g02135_" + STATE.extCon + "_raster_daily_" + STATE.norSouth,
-                SRS: getLocationParams().srs,
-                BBOX: getLocationParams().locationVal,
-                TILED: false,
-                format:"image/png",
-                TIME: STATE.current.format("YYYY-MM-DD"),
-                STYLES: "NSIDC:g02135_" + STATE.extCon + "_raster_basemap"
-            };
-            await updateWMSLayerParams(map.getLayers().getArray()[0],wmsParams);
-            await sleep(STATE.rate);
-        }
-        else {
-            await sleep(50);
-        }
+/** Actual animation loop
+ * @param {map} map - The map object to animate
+*/
+async function animationLoop() {
+  while (true) {
+    if (!STATE.stop) {
+      nextDate();
+      [map, projection] = await getState(map, projection);
+      let sourceType = 'monthly';
+      if (STATE.dateLoopStyle == 'daily') {
+        sourceType = 'daily';
+      }
+      const wmsParams = {
```

Comment:
Create helper method to get WMS parameters instead of reusing the same code in multiple areas.